pkg/network/protocols/postgres/telemetry.go

```diff
@@ -67,31 +58,34 @@ func createQueryLengthBuckets(metricGroup *libtelemetry.MetricGroup) [numberOfBu
 }
 
 // NewTelemetry creates a new Telemetry
-func NewTelemetry() *Telemetry {
+func NewTelemetry(cfg *config.Config) *Telemetry {
 	metricGroup := libtelemetry.NewMetricGroup("usm.postgres")
 
 	return &Telemetry{
 		metricGroup:               metricGroup,
 		queryLengthBuckets:        createQueryLengthBuckets(metricGroup),
 		failedTableNameExtraction: metricGroup.NewCounter("failed_table_name_extraction", libtelemetry.OptStatsd),
 		failedOperationExtraction: metricGroup.NewCounter("failed_operation_extraction", libtelemetry.OptStatsd),
+		telemetryBufferSize:       cfg.MaxPostgresTelemetryBuffer,
 	}
 }
 
 // getBucketIndex returns the index of the bucket for the given query size
-func getBucketIndex(querySize int, options ...CountOptions) int {
-	if len(options) > 0 && options[0].TelemetryBufferSize > ebpf.BufferSize {
-		firstBucketLowerBoundary = options[0].TelemetryBufferSize - numberOfBucketsSmallerThanMaxBufferSize*bucketLength + 1
-	}
+func (t *Telemetry) getBucketIndex(querySize int) int {
+	// firstBucketLowerBoundary is the lower boundary of the first bucket.
+	// We add 1 in order to include BufferSize as the upper boundary of the third bucket.
+	// Then the first three buckets will include query lengths shorter or equal to BufferSize,
+	// and the rest will include sizes equal to or above the buffer size.
+	firstBucketLowerBoundary := t.telemetryBufferSize - numberOfBucketsSmallerThanMaxBufferSize*bucketLength + 1
```

Comment:
seems like it is constant for the entire lifetime of `t`, should be move it to the initializer method?